packages/core/src/sync-historical/index.ts

```diff
@@ -535,13 +536,32 @@ export const createHistoricalSync = async (
       }),
     ).then((traces) => traces.flat());
 
-    if (traces.length > 0) {
-      await args.syncStore.insertTraces({
-        traces,
+    if (isKilled) return;
+
+    await args.syncStore.insertTraces({
+      traces,
+      chainId: args.network.chainId,
+    });
+
+    if (isKilled) return;
+
+    if (shouldGetTransactionReceipt(filter)) {
+      const transactionHashes = new Set(
```

Comment:
Totally speculative, but I wonder if it makes sense to initialize this before we call `insertTraces` so that the relatively large `traces` object can be GC'd sooner.